crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,518 @@
+                    //          = ∆x_{n} + l / (-l')
+                    possible_target_base_delta =
+                        possible_target_base_delta + loss / negative_loss_derivative;
+                }
+            }
+
+            // Final solvency check.
+            if !self
+                .solvency_after_long(
+                    possible_target_base_delta,
+                    self.calculate_open_long(possible_target_base_delta)
+                        .unwrap(),
+                    checkpoint_exposure,
+                )
+                .is_some()
+            {
+                return Err(eyre!("Guess in `get_targeted_long` is insolvent."));
+            }
+
+            // Final accuracy check.
+            let possible_target_bond_delta = self
+                .calculate_open_long(possible_target_base_delta)
+                .unwrap();
+            let resulting_rate =
+                self.rate_after_long(possible_target_base_delta, Some(possible_target_bond_delta));
+            if target_rate > resulting_rate {
+                return Err(eyre!("get_targeted_long: We overshot the zero-crossing.",));
+            }
+            let loss = resulting_rate - target_rate;
+            if !(loss < allowable_error) {
+                return Err(eyre!(
+                    "get_targeted_long: Unable to find an acceptible loss. Final loss = {}.",
+                    loss
+                ));
+            }
+
+            Ok(possible_target_base_delta)
+        }
+    }
+
+    /// The fixed rate after a long has been opened.
+    ///
+    /// We calculate the rate for a fixed length of time as:
+    /// $$
+    /// r(x) = (1 - p(x)) / (p(x) t)
+    /// $$
+    ///
+    /// where $p(x)$ is the spot price after a long for `delta_bonds`$= x$ and
+    /// t is the normalized position druation.
+    ///
+    /// In this case, we use the resulting spot price after a hypothetical long
+    /// for `base_amount` is opened.
+    fn rate_after_long(
+        &self,
+        base_amount: FixedPoint,
+        bond_amount: Option<FixedPoint>,
+    ) -> FixedPoint {
+        let annualized_time =
+            self.position_duration() / FixedPoint::from(U256::from(60 * 60 * 24 * 365));
+        let resulting_price = self.get_spot_price_after_long(base_amount, bond_amount);
+        (fixed!(1e18) - resulting_price) / (resulting_price * annualized_time)
+    }
+
+    /// The derivative of the equation for calculating the rate after a long.
+    ///
+    /// For some $r = (1 - p(x)) / (p(x) * t)$, where $p(x)$
+    /// is the spot price after a long of `delta_base`$= x$ was opened and $t$
+    /// is the annualized position duration, the rate derivative is:
+    ///
+    /// $$
+    /// r'(x) = \frac{(-p'(x) p(x) t - (1 - p(x)) (p'(x) t))}{(p(x) t)^2} //
+    /// r'(x) = \frac{-p'(x)}{t p(x)^2}
+    /// $$
+    ///
+    /// We return $-r'(x)$ because negative numbers cannot be represented by FixedPoint.
+    fn negative_rate_after_long_derivative(
```

Comment:
This may just be me, but I read this as "we are computing the derivative of the negative rate," which is a little confusing. This name `rate_after_long_derivative_negation` would be clearer to me, but I'll let you make this call.